### PR TITLE
Allow JacksonJsonpMapper user provided ObjectMapper to maintain configuration

### PR DESCRIPTION
**Pull request by mluckam**
The JacksonJsonpMapper has a public constructor that takes in a user provided ObjectMapper to allow customization for serialization and deserialization.  The constructor can be misleading since it overrides two configurations to disable pretty print and disregard null values on serialization.  As far as I can tell, there is no other way to modify the ObjectMapper configuration other than through the constructor.

The proposal is that these configurations be moved to the default constructor.  This allows users with no preference to take the default configuration, but if custom configuration is desired a user can provide the ObjectMapper.  A few advantages to this strategy:

1. Default constructor still provides desired default configuration
2. Provides more flexibility of configuration to the user
3. Constructor with parameter ObjectMapper will be less confusing to the users

For context, the change was made in this [commit](https://github.com/elastic/elasticsearch-java/commit/1df3f02ceaa3e1900ff58995eeb6e7a628c207a0#diff-8b6b1783b9732bea958c33e27b933df8bd8f87643315d82cf5a3ba5f8469766c).

### CODE REVIEW

**Comment by mluckam** [comment]
@swallez would you mind providing feedback to this PR?  It appears that you were the author of the constructor constraint in question, see [commit](https://github.com/elastic/elasticsearch-java/commit/1df3f02ceaa3e1900ff58995eeb6e7a628c207a0#diff-8b6b1783b9732bea958c33e27b933df8bd8f87643315d82cf5a3ba5f8469766c).

**Comment by swallez** [comment]
@elasticmachine ok to test

File: java-client/src/main/java/co/elastic/clients/util/Pair.java [hunk 1]
```diff
@@ -56,7 +56,8 @@ public static <K, V> Pair<K, V> of(K name, V value) {
         return new Pair<>(name, value);
     }
 
-    public static <K, V> JsonpDeserializer<Pair<K, V>> deserializer(Function<String, K> keyDeserializer, JsonpDeserializer<V> valueDeserializer) {
+    public static <K, V> JsonpDeserializer<Pair<K, V>> deserializer(Function<String, K> keyDeserializer,
```
Comment (swallez) [inline java-client/src/main/java/co/elastic/clients/util/Pair.java:59]: Please avoid reformatting unrelated files in the PR.

**Review by swallez — COMMENTED** [review]
Thanks @mluckam. Your proposed changes totally make sense!

I've left a few minor comments about code formatting. Can you please fix it?

File: java-client/src/main/java/co/elastic/clients/util/Pair.java (hunk 1 above)
Comment (mluckam) [reply java-client/src/main/java/co/elastic/clients/util/Pair.java]: Looks like the checkstyle finding corrected here was fixed in the following [commit](https://github.com/elastic/elasticsearch-java/commit/323498f8214c03fc0eb8edff9ab3542e2d5e07bf#diff-45e77db90a1ba56a54a38cec1b93c14e49b069b1f11f7cb843adad3c399f38d1R59).  Rebase corrected the issue.

File: java-client/src/main/java/co/elastic/clients/json/jackson/JacksonJsonpMapper.java [hunk 2]
```diff
@@ -45,17 +45,17 @@ private JacksonJsonpMapper(ObjectMapper objectMapper, JacksonJsonProvider provid
...
+        this(objectMapper,
             // Creating the json factory from the mapper ensures it will be returned by JsonParser.getCodec()
             new JacksonJsonProvider(objectMapper.getFactory())
         );
     }
 
     public JacksonJsonpMapper() {
-        this(new ObjectMapper());
+    	this(new ObjectMapper()
+                    .configure(SerializationFeature.INDENT_OUTPUT, false)
```
[the remark this answers is not in the document]
Comment (mluckam) [reply java-client/src/main/java/co/elastic/clients/json/jackson/JacksonJsonpMapper.java]: Updated continuation indent from 8 to 4.

**Review by swallez — APPROVED** [review]
LGTM! Thanks for your contribution!

**Comment by cesarlino** [comment]
It is great to have that implemented. Thanks for that.
I am wondering if this will be in version 7.17.8 and when it is planned to release?

Thanks
clino
